Set up the builders (Meson and Nix) and develoment shell; expand .gitignore and README.md

ShamrockLee opened the pull request:
This pull request (PR) set up the basic environment for library examples.

If applied, we could then put the `zip-ls`, `zip-stat` and `zip-cat` test cases inside the `test/libzip-examples` directory and add the corresponding lines to the `tests/libzip-examples/meson.build` file.

@Lin97140 @Paul-cod please help me test the meson build and test workflow on your system, and see if there's anything in need of correction / explanation before merging this PR.

Here is the piece of GitHub documentation about reviewing a PR:
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews

ShamrockLee committed:
.gitignore: add various ignore rules from github/gitignore

Original: C
Add:
*   Backup (backup files)
*   Emacs
*   Ninja (builder)
*   Nix
*   Vim
*   Visual Studio Code

ShamrockLee committed:
fetch-clang-format: init

ShamrockLee committed:
README.md: update and document the .clang-format fetching

ShamrockLee committed:
.gitignore: add rules for Meson and drop the shadowed Ninja rules

ShamrockLee committed:
Setup builders: Meson and Nix

Paul-cod commented on meson.build:
```diff
@@ -0,0 +1,38 @@
+  use_fuse3 = true
+  cfg_h.set('FUSE_USE_VERSION', 30)
+else
+  dep_libfuse = dep_libfuse2
```
mac上需要再加cfg_h.set('FUSE_USE_VERSION', 25)
```suggestion
  dep_libfuse = dep_libfuse2
  cfg_h.set('FUSE_USE_VERSION', 25)
```

ShamrockLee committed:
meson.build: fix FUSE_USE_VERSION on Mac osxfuse

This adds
```c
#define FUSE_USE_VERSION 25
```
to `config.h` when libfuse2 is used.

Co-authored-by: Paul-cod <75133243+Paul-cod@users.noreply.github.com>

ShamrockLee committed:
Use skeeto/optparse as source bundle

ShamrockLee committed:
src/zipfs-rw.c: init

ShamrockLee committed:
Pass flake self to override the package source

ShamrockLee committed:
flake.nix: use clang-tools for clang-format

ShamrockLee committed:
enable version printing and init include/

ShamrockLee committed:
README.md: correct the Nix flake build instruction
